### PR TITLE
Improved qhull triangulations with large x,y offset

### PR DESCRIPTION
**Pull request by ianthomas23**
Fixes issue #8682.

Rather than changing the qhull options that we use, I am explicitly subtracting the means from the x, y coordinates before calling qhull to calculate the Delaunay triangulation. If you have coordinate positions `offset` and `1+offset`, previously the triangulation would fail (merging 'nearby' points) for `offset=1e5`. Now it is OK for at least `offset=1e15`, but I've set the limit in the test to `offset=1e10` in case different architectures/OSes give different numerical errors.

There are some minor changes to two of the test images, so I have added a API change note that other triangulations may be slightly modified by this change.

### CODE REVIEW

**Comment by eric-wieser** [comment]
Can you add an image comparison test to demonstrate the plotting from #8682?

**Comment by eric-wieser** [comment]
Tests are complaining about `tri_smooth_contouring`

**Comment by tacaswell** [comment]
I am happy with checking that the triangulation is of a consistent size.

We might need to put a small threshold on that test now that we are pushing the precision of the system?  I assume @ianthomas23 you are not on a mac?

**Comment by ianthomas23** [comment]
@tacaswell You are correct, I use Linux.

Another option is to simplify the test script/image. At the moment it is a direct copy of one of the examples and has lots of very closely-spaced contours, some of them overlapping.

**Comment by tacaswell** [comment]
Either option is fine.

File: doc/api/api_changes/2017-07-14-IT_qhull_large_offset.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Improved Delaunay triangulations with large offsets
```
Comment (dstansby) [inline doc/api/api_changes/2017-07-14-IT_qhull_large_offset.rst]: Could you add a mention of the Matplotlib methods that are affected by this change? Users might not be aware of what a Delaunay triangulation is or what Matplotlib is doing under the hood, but will be able to identify with a specific Matplotlib method that they use.

Comment (QuLogic) [inline doc/api/api_changes/2017-07-14-IT_qhull_large_offset.rst]: Ping @ianthomas23.

Comment (ianthomas23) [inline doc/api/api_changes/2017-07-14-IT_qhull_large_offset.rst]: @dstansby @QuLogic My apologies, I thought I'd responded to this ages ago but evidently I didn't. I have amended the API change text now.

**Comment by tacaswell** [comment]
As always, thank you @ianthomas23 !

**Pull request merged**